app/views/applications/show.html.erb

```diff
@@ -13,7 +13,15 @@
   </section>
   <section id="single-pets">
     <% @application.pets.each do |pet| %>
-      <%= link_to pet.name, "/pets/#{pet.id}" %> | <%= link_to "Approve Single Pet", "/applications/#{@application.id}/#{pet.id}", method: "patch" %><br/>
+      <%= link_to pet.name, "/pets/#{pet.id}" %>
+      <p></p>
+      <% if pet.application_approved == @application.id.to_s%>
+        <%= link_to "Revoke #{pet.name} application", "/applications/#{@application.id}/pets/#{pet.id}", method: :put%><br/>
+      <%elsif pet.application_approved == '-1' || pet.application_approved == nil%>
```

Comment:
Might be better to say `if pet.application_approved.nil?` -> revoke, and `else` -> approve